Square.cpp

```diff
@@ -1,15 +1,37 @@
 #ifndef Square_cpp
 #define Square_cpp
+
 #include <iostream>
-#include "Draw.cpp"
+#include "Shape.cpp"
+#include "Area.cpp"
+#include "Input.cpp"
+
 using namespace std;
 
-class Square : public Draw
+class Square : public Shape , public Input
 {
 public:
     void drawShape() override
     {
         cout << "square" << endl;
     }
+
+    int inputLength() override 
+    {
+        int length;
+        cout << "Enter Length" << endl;
+        cin >> length;
+        return length;
+    }
+    int CalculateArea() override
+    {
+        int length = inputLength ();
+        int width = inputLength ();
```

Comment:
Why are we taking input from user for both `length` and `width`?

To calculate a square's area we only need length of one side.

Reply:
Actually we are only taking one input from the user because in square length and width is the same so just to make it easier to understand i mentioned both length and width in the formula

Reply:
You're calling the same function `inputLength` twice. Which means user will input twice the same number. This puts extra burden on the user which is very bad user experience.

Not to mention, there's no way to enforce user enters the same value the 2nd time. The user may enter 2 on first call and 3 on second call.

Reply:
This is still not resolved!